Makefile

```diff
@@ -0,0 +1,32 @@
+# The version needs an annotated tag in the git repo in the form vX.Y.Z.
+# If such is not present, fall back to using v0.0.0 so we get a non-empty
+# version string.
+VERSION=$(shell (git describe --match 'v*' --long 2>/dev/null || echo 'v0.0.0') | sed 's/^v//')
+
+usage:
+	@echo "Try running \`make dist\` to build a tarball"
+
+submodules:
+	git submodule update --init
+
+glass:
+	cd src/glass && npm install && npx ng build --prod --output-hashing=all
```

Comment:
This is my makefile ignorance here, but wouldn't it be better to use `pushd` then `popd` to ensure that the following steps are in the expected dir (eg on line 23)?

(I figure it doesn't actually matter if this is working)

Reply:
I don't think it makes a material difference in this case. The `cd` at the start here only has effect for the duration of that line, it doesn't change the overall environment, i.e. if there were another command in that block, when it was invoked, it'd be back in the original directory automatically.